spec/controllers/application_controller/policy_support_spec.rb

```diff
@@ -17,4 +17,28 @@
       controller.send(:assign_policies, Host)
     end
   end
+
+  describe '#policy_sim' do
+    context 'canceling policy simulation' do
+      before do
+        allow(controller).to receive(:flash_to_session).and_call_original
+        allow(controller).to receive(:previous_breadcrumb_url).and_return('')
+        allow(controller).to receive(:redirect_to)
+        controller.instance_variable_set(:@_params, :button => 'cancel')
+      end
+
+      it 'calls redirect_to method' do
+        expect(controller).to receive(:redirect_to).with('')
+        controller.send(:policy_sim)
+      end
+
+      context 'flash array about canceling policy simulation' do
+        it 'adds the message to flash array' do
+          expect(controller).to receive(:flash_to_session).with('Edit policy simulation was cancelled by the user')
+          controller.send(:policy_sim)
+          expect(controller.instance_variable_get(:@flash_array)).to eq([{:message => 'Edit policy simulation was cancelled by the user', :level => :success}])
```

Comment:
What about putting the string into a (regular) variable so you don't have to write it twice?